test/interpreter_functional/test_suites/run_pipeline/event_annotation/fetch_event_annotations.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { omit } from 'lodash';
+import { ExpectExpression, expectExpressionProvider } from '../helpers';
+import { FtrProviderContext } from '../../../../functional/ftr_provider_context';
+import { expectedResult } from './fetch_event_annotations_result';
+
+export default function ({
+  getService,
+  updateBaselines,
+}: FtrProviderContext & { updateBaselines: boolean }) {
+  let expectExpression: ExpectExpression;
+
+  describe('fetch event annotation tests', () => {
+    before(() => {
+      expectExpression = expectExpressionProvider({ getService, updateBaselines });
+    });
+
+    const timeRange = {
+      from: '2015-09-21T00:00:00Z',
+      to: '2015-09-22T00:00:00Z',
+    };
+
+    it(`manual annotations from different groups`, async () => {
+      const expression = `
+      kibana_context timeRange={timerange from='${timeRange.from}' to='${timeRange.to}'}
+      | fetch_event_annotations timezone="Australia/Darwin"
+        interval="1w" 
+        groups={event_annotation_group   dataView={indexPatternLoad id="logstash-*"}  annotations={
+          manual_point_event_annotation id="ann1" label="Manual1" color="red" icon="bolt" time="2015-09-21T12:15:00Z" lineWidth="3" lineStyle="solid" textVisibility=true
+          } 
+          annotations={
+            manual_point_event_annotation id="ann2" label="ManualHidden" color="pink" icon="triangle" time="2015-09-21T12:30:00Z" isHidden=true
+          }
+        }
+        groups={event_annotation_group dataView={indexPatternLoad id="logstash-*"} annotations={
+          manual_range_event_annotation id="ann3" label="Range" color="blue" time="2015-09-21T07:30:00Z" endTime="2015-09-21T12:30:00Z"
+        }
+      }
+    `;
+      const result = await expectExpression('fetch_event_annotations', expression).getResponse();
+
+      expect(result.rows.length).to.equal(2); // filters out hidden annotations
+      expect(result.rows).to.eql([
+        {
+          id: 'ann3',
+          time: '2015-09-21T07:30:00Z',
+          endTime: '2015-09-21T12:30:00Z',
+          timebucket: '2015-09-20T14:30:00.000Z',
+          type: 'range',
+          label: 'Range',
+          color: 'blue',
+        },
+        {
+          id: 'ann1',
+          time: '2015-09-21T12:15:00Z',
+          timebucket: '2015-09-20T14:30:00.000Z', // time bucket is correctly assigned
```

Comment:
I like these comments on your assertions. Let's me know more about what you're really checking for.